update_version_next.py

```diff
@@ -0,0 +1,92 @@
+#!/usr/bin/env python3
+"""
+Replace `.. versionchanged:: next` lines in docs files by the given version.
+
+Run this at release time to replace `next` with the just-released version
+in the sources.
+
+No backups are made; add/commit to Git before running the script.
+
+Applies to all the VersionChange directives. For deprecated-removed, only
+handle the first argument (deprecation version, not the removal version).
+
+"""
+
+import argparse
+import re
+import sys
+from pathlib import Path
+
+DIRECTIVE_RE = re.compile(
+    r"""
+        (?P<before>
+            \s*\.\.\s+
+            (version(added|changed|removed)|deprecated(-removed)?)
+            \s*::\s*
+        )
+        next
+        (?P<after>
+            .*
+        )
+    """,
+    re.VERBOSE | re.DOTALL,
+)
+
+parser = argparse.ArgumentParser(
+    description=__doc__, formatter_class=argparse.RawDescriptionHelpFormatter
+)
+parser.add_argument(
+    "version",
+    help='String to replace "next" with. Usually `x.y`, but can be anything.',
+)
+parser.add_argument(
+    "directory",
+    type=Path,
+    nargs="?",
+    help="Directory to process",
+)
+parser.add_argument(
+    "--verbose",
+    "-v",
+    action="count",
+    default=0,
+    help="Increase verbosity. Can be repeated (`-vv`).",
+)
+
+
+def main(argv: list[str]) -> None:
+    args = parser.parse_args(argv)
+    version = args.version
```

Comment:
Shall we validate that the input is of the form "x.y"?

Reply:
I'd just trust the caller here.